Navigation styling

rajatvijay opened the pull request:
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #1666

<!-- Concisely describe what the pull request does. -->

Separate Issues
1. https://github.com/centerofci/mathesar/issues/2027
2. https://github.com/centerofci/mathesar/issues/2029

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

seancolsen reviewed (changes requested):
This is looking good, and very close to merge-ready! 

Changes requested:

- On a URL like `http://localhost:8000/mathesar_tables/217/explorations/edit/5/` the breadcrumb structure doesn't match Figma. Can you fix this?

    | Figma | App |
    | -- | -- |
    | ![image](https://user-images.githubusercontent.com/42411/206218022-94dddfd2-ab5b-4924-8ccc-420940dd71a9.png) | ![image](https://user-images.githubusercontent.com/42411/206217485-a382f933-209c-4728-9377-093a86fbac1f.png) |

- When you fix the conflicts that this PR now has, please make sure to test a record page like this:

    ![image](https://user-images.githubusercontent.com/42411/206248007-b65a07ab-2ea6-4043-8d5c-45fbfce91003.png)

    Here the record summary has a `NULL` value inside it. I made changes in #2017 to stylize those values (hence the conflicts in your PR). Please make sure that the record summary looks good inside the navigation header now that it'll have a `NULL` value in it. If you run into problems you can ping me.

Notes

- In bb0650ad7b03960601d165fed198def9a55d49eb I fixed the issue with the logo. The SVG did actually have the white outline in it, but it was using an SVG `<mask>` element, which is [not well supported](https://caniuse.com/mdn-svg_elements_mask). Further, the actual SVG data was pretty messy with lots of superfluous path segments and nodes. I used Inkscape to clean it up, producing an SVG with only two (relatively simple) paths. I ended up with something _much_ smaller which I think is now suitable for inlining (hence the svelte file instead of svg file).

- I added an agenda item to our [front end team agenda](https://hackmd.io/ctEvo2tzQTyFV_38noSSEQ?edit) _"Refactoring ButtonMenuItem to use an onClick prop instead of a click event"_. This is to suggest an alternate approach to `ButtonMenuItem` that would lead to cleaner code inside the `AppHeader` you were just working on (because we wouldn't need to track the loading state in `AppHeader` anymore).

rajatvijay commented:
> This is looking good, and very close to merge-ready!
> 
> Changes requested:
> 
>     * On a URL like `http://localhost:8000/mathesar_tables/217/explorations/edit/5/` the breadcrumb structure doesn't match Figma. Can you fix this?
>       
>       
>       
>       Figma
>       App
>       
>       
>       
>       
>       ![image](https://user-images.githubusercontent.com/42411/206218022-94dddfd2-ab5b-4924-8ccc-420940dd71a9.png)
>       ![image](https://user-images.githubusercontent.com/42411/206217485-a382f933-209c-4728-9377-093a86fbac1f.png)

76932adb

>     * When you fix the conflicts that this PR now has, please make sure to test a record page like this:
>       ![image](https://user-images.githubusercontent.com/42411/206248007-b65a07ab-2ea6-4043-8d5c-45fbfce91003.png)
>       Here the record summary has a `NULL` value inside it. I made changes in #2017 to stylize those values (hence the conflicts in your PR). Please make sure that the record summary looks good inside the navigation header now that it'll have a `NULL` value in it. If you run into problems you can ping me.

edda9c2ed1c781cb33d0804f75159fec8e138e81

> 
> Notes
> 
>     * In [bb0650a](https://github.com/centerofci/mathesar/commit/bb0650ad7b03960601d165fed198def9a55d49eb) I fixed the issue with the logo. The SVG did actually have the white outline in it, but it was using an SVG `<mask>` element, which is [not well supported](https://caniuse.com/mdn-svg_elements_mask). Further, the actual SVG data was pretty messy with lots of superfluous path segments and nodes. I used Inkscape to clean it up, producing an SVG with only two (relatively simple) paths. I ended up with something _much_ smaller which I think is now suitable for inlining (hence the svelte file instead of svg file).
> 
>     * I added an agenda item to our [front end team agenda](https://hackmd.io/ctEvo2tzQTyFV_38noSSEQ?edit) _"Refactoring ButtonMenuItem to use an onClick prop instead of a click event"_. This is to suggest an alternate approach to `ButtonMenuItem` that would lead to cleaner code inside the `AppHeader` you were just working on (because we wouldn't need to track the loading state in `AppHeader` anymore).

rajatvijay commented on mathesar_ui/src/components/breadcrumb/BreadcrumbSelector.svelte:
```diff
@@ -25,17 +31,20 @@
   >
-    <BreadcrumbSeparatorIcon />
-  </button>
+    <div class="trigger">
```
c358a4c4

rajatvijay commented on mathesar_ui/src/components/breadcrumb/BreadcrumbRecordSelector.svelte:
```diff
@@ -0,0 +1,44 @@
+</script>
+
+<Button on:click={handleClick} appearance="ghost" class="padding-zero">
+  <div class="trigger">
```
c358a4c4

seancolsen reviewed (approved):
Looks good. There is a small problem with `NULL` values within the record summary. I opened #2055 to handle it later.
